gateway/scaling/function_scaler.go

```diff
@@ -106,3 +139,23 @@ func (f *FunctionScaler) Scale(functionName string) FunctionScaleResult {
 		Duration:  time.Since(start),
 	}
 }
+
+type routine func(attempt int) error
+
+func backoff(r routine, attempts int, interval time.Duration) error {
```

Comment:
we should really consider adding `ctx` to this and checking that the `ctx` is cancelled/done and also passing it to the routine so that it can also be used to cancel work or to support tracing. Context is generally most useful for methods that talk to external services, which is a big part of why `backoff` exists. It is a natural place to put `ctx` and even if we are not passing it around everywhere, it is clearly a place it would be wanted in the future.

Reply:
Whilst `Context` objects are used for this often in Golang code, the context is not consumable by any of the APIs we're calling. It's for that reason that I can't see how this would help right now. Perhaps you would be interested in doing some refactoring on the APIs to enable it to be consumed after this PR is merged?

My intent with `backoff()` is to resolve the current issue, not to create a public package for general/generic use, but if this pattern is useful I think we could refactor it and extract it later.

Reply:
ok

Reply:
I don't want it to sound like I'm not listening to your idea. I spent a couple of hours +/- trying to retrofit Context into the current usage. It didn't seem to make sense with the underlying APIs.

Reply:
I understand